src/Umbraco.New.Cms.Infrastructure/Factories/Installer/DatabaseSettingsFactory.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.Extensions.Options;
+using Umbraco.Cms.Core.Configuration.Models;
+using Umbraco.Cms.Core.Mapping;
+using Umbraco.Cms.Infrastructure.Persistence;
+using Umbraco.Extensions;
+using Umbraco.New.Cms.Core.Factories;
+using Umbraco.New.Cms.Core.Models.Installer;
+
+namespace Umbraco.New.Cms.Infrastructure.Factories.Installer;
+
+public class DatabaseSettingsFactory : IDatabaseSettingsFactory
+{
+    private readonly IEnumerable<IDatabaseProviderMetadata> _databaseProviderMetadata;
+    private readonly IOptionsMonitor<ConnectionStrings> _connectionStrings;
+    private readonly IUmbracoMapper _mapper;
+
+    public DatabaseSettingsFactory(
+        IEnumerable<IDatabaseProviderMetadata> databaseProviderMetadata,
+        IOptionsMonitor<ConnectionStrings> connectionStrings,
+        IUmbracoMapper mapper)
+    {
+        _databaseProviderMetadata = databaseProviderMetadata;
+        _connectionStrings = connectionStrings;
+        _mapper = mapper;
+    }
+
+    /// <inheritdoc/>
+    public ICollection<DatabaseSettingsModel> GetDatabaseSettings()
+    {
+        ConnectionStrings? connectionString = _connectionStrings.CurrentValue;
+
+        // If the connection string is configured we just return the configured provider.
+        if (connectionString.IsConnectionStringConfigured())
+        {
+            var providerName = connectionString.ProviderName;
+            if (providerName is null)
+            {
+                throw new InvalidOperationException(
+                    "Database provider must be configured to use preconfigured connection string");
+            }
+
+            IDatabaseProviderMetadata? providerMetaData = _databaseProviderMetadata
+                .FirstOrDefault(x => x.ProviderName?.Equals(providerName, StringComparison.InvariantCultureIgnoreCase) ?? false);
+
+            if (providerMetaData is null)
+            {
+                throw new InvalidOperationException($"Provider {providerName} is not a registered provider");
+            }
+
+            DatabaseSettingsModel configuredProvider = _mapper.Map<DatabaseSettingsModel>(providerMetaData)!;
```

Comment:
This mapping does not take into account a custom named preconfigured default database. If I have a preconfigured connection string like this: 

`"Data Source=|DataDirectory|/ThisIsMyDbName.sqlite.db;Cache=Shared;Foreign Keys=True;Pooling=True"`

... the mapped response model still contains the default DB name:

```json
"databases": [
	{
		"id": "530386a2-b219-4d5f-b68c-b965e14c9ac9",
		"sortOrder": -1,
		"displayName": "SQLite",
		"defaultDatabaseName": "Umbraco",
		"providerName": "Microsoft.Data.Sqlite",
		"isConfigured": true,
		"requiresServer": false,
		"serverPlaceholder": "",
		"requiresCredentials": false,
		"supportsIntegratedAuthentication": false,
		"requiresConnectionTest": false
	}
]
```

...but ultimately the installed database will still adhere to my preconfigured DB name (ThisIsMyDbName) when the installer runs.

Reply:
That's because it's the `IDatabaseProviderMetadata` that's mapped, so essentially this doesn't represent what's currently configured, but how, in this case, SQLite is by default, if that makes sense? 

So in this case "Umbraco" is still the default name for SQLite, even though your specific instance is configured as "ThisIsMyDbName" if that makes sense?

Reply:
I get the mapping part. I just think maybe it would be prudent to reflect back to the installer UI what will actually happen when the installation runs - i.e. creating a DB with another name than the default one. 

This of course is assuming that the UI plans on displaying the DB name to the user.

Maybe we can just wait with this and fix it later - if it needs fixing (and definitively do so outside of the mapper :smile: ).

Reply:
Ahh yeah that is a very good point indeed, but I agree, I think it's best to wait and see if it's something the UI is actually gonna need, and then we can look at it then :) and for now just keep the behavior the same is the "old" installer :) 

And yes please, the new mappers are so pretty and simple, let's keep it that way haha 🤣